File: Go/json.go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"os"
)

func main() {
	// Encode data from a map to JSON
	type ColorGroup struct {
		ID     int
		Name   string
		Colors []string
	}
	group := ColorGroup{
		ID:     1,
		Name:   "Reds",
		Colors: []string{"Crimson", "Red", "Ruby", "Maroon"},
	}
	currentJson, err := json.Marshal(group)
	if err != nil {
		log.Println(err)
	}
	fmt.Println(string(currentJson))
	// Write data to a file
	err = writeContnetToFile("test.json", string(currentJson))
	if err != nil {
		log.Println(err)
	}
	// Check if a json file is valid.
	validCheck, err := validateJsonFromFile("test.json")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(validCheck)
}

// Open a file and check if a json is valid.
func validateJsonFromFile(jsonFile string) (bool, error) {
	data, err := os.ReadFile(jsonFile)
	if err != nil {
		return false, err
	}
	return json.Valid(data), nil
}

// Check if a json is valid.
func validateJson(jsonData []byte) bool {
	return json.Valid(jsonData)
}

// Don't append and write to file
func writeContnetToFile(filepath string, content string) error {
	err := os.WriteFile(content, []byte(filepath), 0644)
	if err != nil {
		return err
	}
	return nil
}

File: JavaScript/json.js
const fs = require("fs")

function main() {
    var sampleJson = `{
    "names": [
        {
            "first": "John",
            "last": "Doe"
        },
        {
            "first": "Jane",
            "last": "Doe"
        }
    ]
}`
    console.log(sampleJson)
    // Parse the json and return it.
    console.log(parseJson(sampleJson))
    // Validate the json and return true or false.
    console.log(validateJson(sampleJson))
    // Open a file and check if the json is valid
    console.log(checkIfFileIsValidJson("node.json"))
}

main()

// Parse the json and return it.
function parseJson(json) {
    return JSON.parse(json)
}

// Validate the json and return true or false.
function validateJson(json) {
    try {
        JSON.parse(json)
        return true
    } catch (e) {
        return false
    }
}

// Open a file and check if it is a valid json.
function checkIfFileIsValidJson(filePath) {
    var json = fs.readFileSync(filePath, "utf8")
    return validateJson(json)
}
